test/RenPool.test.js

```diff
@@ -323,11 +332,118 @@ describe('RenPool contract test', function () {
 
       await renPool.transferRewardsToDarknodeOwner([renBTCAddr]);
+      // ^ OBSERVATION: not sure if the above code is actually doing anything,
+      // we need a way to query the darknode's balance and make sure the
+      // balance is actually being transferred.
+      // Also, not sure if darknodePayment contract is still being used
+      // or has been replaced by the RenVM.
+    });
+
+  });
+
+  describe('deregisterDarknode', function () {
+
+    it('should deregister darknode', async function () {
+      await renToken.connect(bob).approve(renPool.address, POOL_BOND);
+      await renPool.connect(bob).deposit(POOL_BOND);
+
+      await renPool.connect(nodeOperator).approveBondTransfer();
+      await renPool.connect(nodeOperator).registerDarknode(NODE_ID, PUBLIC_KEY);
+
+      await increaseMonth();
+      await darknodeRegistry.connect(alice).epoch();
+
+      expect(await darknodeRegistry.connect(alice).isRegistered(NODE_ID)).to.be.true;
+
+      await renPool.connect(nodeOperator).deregisterDarknode();
+
+      expect(await darknodeRegistry.connect(alice).isPendingDeregistration(NODE_ID)).to.be.true;
+
+      await increaseMonth();
+      await darknodeRegistry.connect(alice).epoch();
+
+      expect(await darknodeRegistry.connect(alice).isDeregistered(NODE_ID)).to.be.true;
+
+      expect(await renPool.connect(owner).darknodeID()).to.equalIgnoreCase(NODE_ID);
+      expect(await renPool.connect(owner).publicKey()).to.equalIgnoreCase(PUBLIC_KEY);
+    });
+
+    it('should fail when darknode deregistration is not performed by node operator', async () => {
+      expect(alice).to.not.equal(nodeOperator);
+
+      await renToken.connect(bob).approve(renPool.address, POOL_BOND);
+      await renPool.connect(bob).deposit(POOL_BOND);
+
+      await renPool.connect(nodeOperator).approveBondTransfer();
+      await renPool.connect(nodeOperator).registerDarknode(NODE_ID, PUBLIC_KEY);
+
+      await increaseMonth();
+      await darknodeRegistry.connect(alice).epoch();
+
+      await expect(
+        renPool.connect(alice).deregisterDarknode()
+      ).to.be.revertedWith('RenPool: Caller is not owner nor node operator');
+
+      expect(await darknodeRegistry.connect(alice).isRegistered(NODE_ID)).to.be.true;
+      expect(await renPool.connect(owner).darknodeID()).to.equalIgnoreCase(NODE_ID);
+      expect(await renPool.connect(owner).publicKey()).to.equalIgnoreCase(PUBLIC_KEY);
     });
 
-    it('should convert base58 to hex', function () {
-      expect(base58ToHex('8MHJ9prQt7UGupfZKSMVes3VzPrGBB'))
-        .to.equalIgnoreCase('0x597869E66F904F741Bf16788F1FCAe36E603F112');
+  });
+
+  describe('refundBond', function () {
+
+    it('should refund all stakers', async () => {
+      const registryStoreBalance = await renToken.connect(owner).balanceOf(darknodeRegistryStoreAddr);
+      const aliceBalance = await renToken.connect(alice).balanceOf(alice.address);
+
+      // Lock pool
+      const deposit = POOL_BOND
```

Comment:
@acuarica I know you removed all comments, but I think it'll help in the future if someone needs/wants to 'audit' the code. I'd add all comments back into the tests